File: algorithmic_toolbox/week3_greedy_algorithms/5_maximum_number_of_prizes/different_summands.py
import sys


def optimal_summands(n: int) -> list[int]:
    summands: list[int] = []

    for i in range(1, n + 1):
        if n - i > i:
            summands.append(i)
            n -= i
        else:
            summands.append(n)
            break

    return summands


if __name__ == "__main__":
    n = int(sys.stdin.read())
    summands = optimal_summands(n)
    print(len(summands))
    for x in summands:
        print(x, end=" ")

File: algorithmic_toolbox/week3_greedy_algorithms/5_maximum_number_of_prizes/different_summands.go
package main

import (
	"C"
	"fmt"
)

func optimalSummands(n int) []int {
	var summands []int

	for i := 1; i <= n; i++ {
		if n-i > i {
			summands = append(summands, i)
			n -= i
		} else {
			summands = append(summands, n)
			break
		}
	}

	return summands
}

func main() {
	var n int
	fmt.Scan(&n)

	summands := optimalSummands(n)

	fmt.Println(len(summands))
	for _, s := range summands {
		fmt.Printf("%d ", s)
	}
}
